onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -177,10 +178,12 @@ static Status GetCapabilityForEP(const GetCapabilityForEPParams& params) {
   }
 
 #if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
-  // Run layout transformer only for EPs other than CPU EP and provided the preferred layout is NHWC
-  // CPU EP layout transformation happens later when level 3 transformers are run.
-  if (params.mode != GraphPartitioner::Mode::kAssignOnly &&
-      current_ep.GetPreferredLayout() == DataLayout::NHWC) {
+  // Run layout transformation for all EPs.
+  // For an EP that wants NHWC this will wrap layout sensitive nodes with Transpose nodes first.
+  // In both NCHW and NHWC EPs the EP specific transpose optimization is run last to optimize
+  // transposes for nodes assigned to the EP or unassigned nodes. This allows things like the
+  // EP aware Resize handling to be run.
+  if (params.mode != GraphPartitioner::Mode::kAssignOnly && params.transform_layout.get()) {
```

Comment:
This feels a little too heavy at it only needs to be called for NCHW EPs with layout sensitive Resize currently. 

The second call to GetCapability also muddies the water a little. Previously the EP could optimize and ignore assigned nodes (although this wasn't required and an EP could ask for an assigned node). With the TryAssignNodes the EP implementation of GetCapability now needs to process at a minimum all unassigned nodes or nodes assigned to the EP. This matters for fusion and compilation as we need complete lists of capabilities from the second call. If the EP ignored assigned nodes we potentially end up in a bad state.

Still doesn't feel like we have the best solution.